ui/src/SmsFulfilment.js

```diff
@@ -0,0 +1,167 @@
+import React, { Component } from "react";
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  TextField,
+} from "@material-ui/core";
+import { getSmsFulfilmentTemplates } from "./Utils";
+
+class SmsFulfilment extends Component {
+  state = {
+    packCode: "",
+    allowableSmsFulfilmentTemplates: [],
+    packCodeValidationError: false,
+    packCodeValidationErrorDesc: "",
+    showDialog: false,
+    phoneNumber: "",
+    newValueValidationError: "",
+    validationError: false,
+  };
+
+  openDialog = () => {
+    this.refreshDataFromBackend();
+    this.setState({
+      showDialog: true,
+    });
+  };
+
+  closeDialog = () => {
+    this.setState({
+      packCode: "",
+      allowableSmsFulfilmentTemplates: [],
+      packCodeValidationError: false,
+      showDialog: false,
+      newValueValidationError: "",
+      phoneNumber: "",
+      validationError: false,
+    });
+  };
+
+  onChangeValue = (event) => {
+    this.setState({
+      phoneNumber: event.target.value,
+    });
+  };
+  refreshDataFromBackend = async () => {
+    const fulfilmentPrintTemplates = await getSmsFulfilmentTemplates(
+      this.props.surveyId
+    ).then((smsTemplates) => {
+      return smsTemplates;
+    });
+
+    this.setState({
+      allowableSmsFulfilmentTemplates: fulfilmentPrintTemplates,
+    });
+  };
+  onSmsTemplateChange = (event) => {
+    this.setState({ packCode: event.target.value });
+  };
+
+  onCreate = async () => {
+    if (!this.state.packCode) {
+      this.setState({
+        packCodeValidationError: true,
+      });
+      return;
+    }
+    if (!this.state.phoneNumber) {
```

Comment:
in ROps we will probably do validation on the client side. I'm OK to skip that for support tool, but we could put it in if we think we've got some code which we'd use for ROps, to get it tested early

Reply:
Going to leave it for now